### ISSUE TITLE
ImportError: cannot import name 'BaseVocab' from 'gguf'

### ISSUE THREAD

**Issue by INZA111** [ggerganov/llama.cpp#7776]
I want to convert LLaVAMistral model to GGUF, I am using tthis code **convert-legacy-llama.py** which is present inside **examples** folder. But got this error **ImportError: cannot import name 'BaseVocab' from 'gguf'**. I didn't found anything online

**Comment by teleprint-me** [comment]
@ggerganov Not sure when, but will look into this. I use this model occassionaly. Support for vision models was temporarily removed. Not sure what the current status is.

**Comment by INZA111** [comment]
Yeah, I am following this ReadMe, I am trying to convert **llava-1.6-mistral-7b** model to gguf but when I am using **examples/convert-legacy-llama.py** I am getting this error **ImportError: cannot import name 'BaseVocab' from 'gguf'**
![llama](https://github.com/ggerganov/llama.cpp/assets/62800000/6271ff01-c1a5-4362-ac72-6209f1a53941)

**Comment by CISC** [comment]
For some reason it's not picking up the repo gguf-py, but using the outdated pip installed one instead, try the following in your venv and run it again:
```bash
pip uninstall gguf
```

**Comment by INZA111** [comment]
Tried **pip uninstall gguf** as well but still facing the same issue
![llama1](https://github.com/ggerganov/llama.cpp/assets/62800000/ae23e1a2-6655-4ea0-9d82-f6c2b7069f4a)

**Comment by CISC** [comment]
Nope, not the same issue. :) Now it can't find `gguf` at all, which confirms my suspicion.

Does `~/llama-gguf/llama-cpp-python/vendor/llama.cpp/gguf-py/gguf/` exist?

**Comment by INZA111** [comment]
yeah this exists 
![gguf](https://github.com/ggerganov/llama.cpp/assets/62800000/1347da0f-2ab2-4119-bcfc-93cac1ad5932)

**Comment by CISC** [comment]
Do you have `NO_LOCAL_GGUF` set?
```bash
echo "${NO_LOCAL_GGUF:?}"
```

**Comment by INZA111** [comment]
Thanks it worked, there was an issue with the path in NO_LOCAL_GGUF

**Comment by CISC** [comment]
Well, it shouldn't be set at all unless you don't want to use the repo `gguf` for some reason?

**Comment by INZA111** [comment]
Okay, thanks for the help :)
